Improve e2e cleanup (#1286)
metalbear-co/mirrord: TESTING.md

MIRRORD_TESTS_USE_BINARY=../target/universal-apple-darwin/debug/mirrord cargo test -p tests
```

### Cleanup

> The Kubernetes resources created by an E2E test are automatically deleted when the test exists successfully. However, failed tests by default don't delete their resources to allow debugging. You can force resource deletion by setting a `MIRRORD_E2E_FORCE_CLEANUP` variable to any value.

Reviewer: ```suggestion
The Kubernetes resources created by the E2E tests are automatically deleted when the test exits successfully. However, failed tests by default don't delete their resources to allow debugging. You can force resource deletion even for failed tests by setting the `MIRRORD_E2E_FORCE_CLEANUP` variable to any value.
```
